reports-scheduler/build.gradle

```diff
@@ -40,6 +40,7 @@ buildscript {
         classpath "${opensearch_group}.gradle:build-tools:${opensearch_version}"
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${kotlin_version}"
         classpath "org.jetbrains.kotlin:kotlin-allopen:${kotlin_version}"
+        classpath "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.22.0-RC2"
```

Comment:
Do we want to add a release candidate library here? @rupal-bq @joshuali925

Reply:
Sorry missed this earlier in the original PR.

Reply:
Alternatively, we can use Detekt v1.21.0 and force use snakeyaml 1.32.